aspnetcore/signalr/configuration.md

```diff
@@ -12,7 +12,401 @@ uid: signalr/configuration
+| `SupportedProtocols` | All installed protocols | Protocols supported by this hub. By default, all protocols registered on the server are allowed, but protocols can be removed from this list to disable specific protocols for individual hubs. |
+| `EnableDetailedErrors` | `false` | If `true`, detailed exception messages are returned to clients when an exception is thrown in a Hub method. The default is `false`, as these exception messages can contain sensitive information. |
+| `StreamBufferCapacity` | `10` | The maximum number of items that can be buffered for client upload streams. If this limit is reached, the processing of invocations is blocked until the server processes stream items.|
+| `MaximumReceiveMessageSize` | 32 KB | Maximum size of a single incoming hub message. |
+| `MaximumParallelInvocationsPerClient` | 1 | The maximum number of hub methods that each client can call in parallel before queueing. |
+
+Options can be configured for all hubs by providing an options delegate to the `AddSignalR` call in `Startup.ConfigureServices`.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddSignalR(hubOptions =>
+    {
+        hubOptions.EnableDetailedErrors = true;
+        hubOptions.KeepAliveInterval = TimeSpan.FromMinutes(1);
+    });
+}
+```
+
+Options for a single hub override the global options provided in `AddSignalR` and can be configured using <xref:Microsoft.Extensions.DependencyInjection.SignalRDependencyInjectionExtensions.AddHubOptions*>:
+
+```csharp
+services.AddSignalR().AddHubOptions<ChatHub>(options =>
+{
+    options.EnableDetailedErrors = true;
+});
+```
+
+### Advanced HTTP configuration options
+
+Use `HttpConnectionDispatcherOptions` to configure advanced settings related to transports and memory buffer management. These options are configured by passing a delegate to [MapHub\<T>](/dotnet/api/microsoft.aspnetcore.builder.hubendpointroutebuilderextensions.maphub) in `Startup.Configure`.
+
+```csharp
+public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+{
+    app.UseRouting();
+
+    app.UseEndpoints(endpoints =>
+    {
+        endpoints.MapHub<ChatHub>("/chathub", options =>
+        {
+            options.Transports =
+                HttpTransportType.WebSockets |
+                HttpTransportType.LongPolling;
+        });
+    });
+}
+```
+
+The following table describes options for configuring ASP.NET Core SignalR's advanced HTTP options:
+
+| Option | Default Value | Description |
+| ------ | ------------- | ----------- |
+| `ApplicationMaxBufferSize` | 32 KB | The maximum number of bytes received from the client that the server buffers before applying backpressure. Increasing this value allows the server to receive larger messages more quickly without applying backpressure, but can increase memory consumption. |
```

Comment:
```suggestion
| `ApplicationMaxBufferSize` | 32 KB | The maximum number of bytes received from the client that the server buffers before applying back pressure. Increasing this value allows the server to receive larger messages more quickly without applying back pressure, but can increase memory consumption. |
```

Should we link to https://wikipedia.org/wiki/Backpressure_routing ? I'm not sure how well "back pressure" will MT. Non english customers tend to go to the english doc for mistranslated words like this.

Reply:
Should we also just update the defaults to 64KB here while we're at it? Also can we put `ApplicationMaxBufferSize` and `TransportMaxBufferSize` next to each other in the table?

Reply:
backpressure routing isn't the same concept here. Surprisingly, I can't find any good links talking about backpressure. https://medium.com/@jayphelps/backpressure-explained-the-flow-of-data-through-software-2350b3e77ce7 explains the concept. I don't think "back pressure" is any better. MT would translate both words separately I assume and that wouldn't make any sense, effectively it could look like you're talking about a massage or something poking you in the back.



> Should we also just update the defaults to 64KB here while we're at it? Also can we put `ApplicationMaxBufferSize` and `TransportMaxBufferSize` next to each other in the table?

That's in RC1, so I'll open another PR for RC1 updates.

Reply:
In that case 1 word, backpressure is better.